### DISCUSSION TITLE
Move more cmake scripts to separate port

### DISCUSSION THREAD

**Discussion by reitowo** [microsoft/vcpkg#29059] [Ideas]
I'm confused about which one should I use: `vcpkg_configure_gn` or `vcpkg_gn_configure` in `vcpkg-gn` port.

In fact, I think most of cmake script should be moved to separate ports, and mark the original script deprecated. Because a single change in cmake scripts will trigger CI to rebuilt almost all ports, regardless of whether port uses the specific changed cmake script.

**Comment by dg0yt** [comment]
You raised a similar question in #29057, so here is a similar answer:
The script port is the future, but currently lacks test coverage and improvements. 

> Because a single change in cmake scripts will trigger CI to rebuilt almost all ports, regardless of whether port uses the specific changed cmake script.

This is wrong. Changes to `vcpkg_configure_gn.cmake` only affect ports which use that function.
The problem with the file is that it cannot be used as a versioned dependency, limiting the scope of potental changes.

**Reply by reitowo** [reply]
Sorry for similar question, I think it could be applied to more scripts so I lifted it to discussion
